services/clusters-keeper/tests/unit/conftest.py

```diff
@@ -209,10 +209,12 @@ def enabled_rabbitmq(
 
 
 @pytest.fixture
-async def initialized_app(app_environment: EnvVarsDict) -> AsyncIterator[FastAPI]:
+async def initialized_app(
+    app_environment: EnvVarsDict, is_pdb_enabled: bool
+) -> AsyncIterator[FastAPI]:
     settings = ApplicationSettings.create_from_envs()
     app = create_app(settings)
-    async with LifespanManager(app, shutdown_timeout=20):
+    async with LifespanManager(app, shutdown_timeout=None if is_pdb_enabled else 20):
```

Comment:
Strangely I never have issues with this when running pytest with debugpy (directly through vs code).

Reply:
@bisgaard-itis  if you stop too much while debugging (e.g. you hit a breakpoint and start  watching data) then it is very normal that the clients (or in this case the lifespanmanager) times out. Now, with this trick, when you pass ``--pdb`` the fixture `is_pdb_enabled=True` and all timeouts are disabled. This way you can "debug without the stress of fearing timing out! ;-)